optaplanner-examples/src/main/java/org/optaplanner/examples/flightcrewscheduling/solver/move/FlightCrewSequentialPillarSwapMoveIteratorFactory.java

```diff
@@ -0,0 +1,103 @@
+import java.util.Random;
+import java.util.SortedSet;
+
+import org.optaplanner.core.impl.heuristic.move.Move;
+import org.optaplanner.core.impl.heuristic.selector.move.factory.MoveIteratorFactory;
+import org.optaplanner.core.impl.score.director.ScoreDirector;
+import org.optaplanner.examples.cheaptime.solver.move.CheapTimePillarSlideMove;
+import org.optaplanner.examples.flightcrewscheduling.domain.Employee;
+import org.optaplanner.examples.flightcrewscheduling.domain.FlightAssignment;
+import org.optaplanner.examples.flightcrewscheduling.domain.FlightCrewSolution;
+
+public class FlightCrewSequentialPillarSwapMoveIteratorFactory implements MoveIteratorFactory<FlightCrewSolution> {
+
+    @Override
+    public long getSize(ScoreDirector<FlightCrewSolution> scoreDirector) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public Iterator<Move<FlightCrewSolution>> createOriginalMoveIterator(ScoreDirector<FlightCrewSolution> scoreDirector) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public RandomFlightCrewSequentialPillarSwapMoveIterator createRandomMoveIterator(
+            ScoreDirector<FlightCrewSolution> scoreDirector, Random workingRandom) {
+        FlightCrewSolution solution = scoreDirector.getWorkingSolution();
+        return new RandomFlightCrewSequentialPillarSwapMoveIterator(solution.getEmployeeList(), workingRandom);
+    }
+
+    public static class RandomFlightCrewSequentialPillarSwapMoveIterator implements Iterator<FlightCrewSequentialPillarSwapMove> {
+
+        private final List<Employee> employeeList;
+        private final Random workingRandom;
+
+        public RandomFlightCrewSequentialPillarSwapMoveIterator(List<Employee> employeeList, Random workingRandom) {
+            this.employeeList = employeeList;
+            this.workingRandom = workingRandom;
+        }
+
+        @Override
+        public boolean hasNext() {
+            return employeeList.size() >= 2;
+        }
+
+        @Override
+        public FlightCrewSequentialPillarSwapMove next() {
+            // TODO once we support overconstrained planning (nullable = true),
+            // it should be possible to select leftEmployee as null and the flightAssignmentSet assigned to null
+            int leftEmployeeIndex = workingRandom.nextInt(employeeList.size());
+            int rightEmployeeIndex = workingRandom.nextInt(employeeList.size() - 1);
+            if (rightEmployeeIndex >= leftEmployeeIndex) {
+                rightEmployeeIndex++;
+            }
+            Employee leftEmployee = employeeList.get(leftEmployeeIndex);
+            Employee rightEmployee = employeeList.get(rightEmployeeIndex);
+            List<FlightAssignment> leftFlightAssignmentList = selectSequentialSubPillar(leftEmployee.getFlightAssignmentSet(), 2);
+            List<FlightAssignment> rightFlightAssignmentList = selectSequentialSubPillar(rightEmployee.getFlightAssignmentSet(), 0);
```

Comment:
Why passing 0 as a `minimumSize`? The move will degrade into pillar change if one of the lists is empty.

Reply:
every one of these move is also a pillar change or swap move. The difference is that this factory selects a subset of those: subpillars that have sequential elements.

Reply:
Ok, I was a bit concerned about the fact this is a swap move, so was not sure if the change move should be permitted too.